snakemake_storage_plugin_sftp/__init__.py

```
@@ -142,157 +114,384 @@
+                "sftp": sftp
+            }
 
+        return self.conn_pool[key]["sftp"]
+
+    def cleanup(self):
+        """Close all connections when the provider is no longer needed."""
+        for conn in self.conn_pool.values():
+            try:
+                conn["sftp"].close()
+                conn["ssh"].close()
+            except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)